test/client/actions/admin_scripts.test.js

```diff
@@ -84,21 +102,97 @@ describe('script actions', () => {
+
     afterEach(() => {
       mock.reset();
     });
+
     describe('axios GET request: ', () => {
       it('should add the appropriate action to the store: ', () => {
         const expectedAction = setScriptsList(scriptListFixtures);
+
         return store.dispatch(fetchAllScripts())
           .then(() => {
             const dispatchedActions = store.getActions();
             const { type, payload } = dispatchedActions[0];
+
             expect(dispatchedActions[0]).toEqual(expectedAction);
             expect(type).toBe('SET_SCRIPTS');
             expect(payload).toEqual(scriptListFixtures);
           });
       });
     });
   });
+
+  describe('setScriptQuestions: ', () => {
+    const setScriptQuestionsResult = setScriptQuestions(questionListFixtures);
+
+    describe('type: ', () => {
+      const { type } = setScriptQuestionsResult;
+
+      it('should have a type property (not undefined): ', () => {
+        expect(type).toBeDefined();
+      });
+      it('should have the type property "SET_SCRIPT_QUESTIONS"', () => {
+        expect(type).toEqual('SET_SCRIPT_QUESTIONS');
+      });
+      it('should not be null: ', () => {
+        expect(type === null).toBe(false);
+      });
+    });
+
+    describe('payload: ', () => {
+      const { payload } = setScriptQuestionsResult;
+
+      it('should have a payload that is an array of objects: ', () => {
+        expect(Array.isArray(payload)).toBe(true);
+      });
+
+      it('should contain 4 objects', () => {
+        const firstObjectProps = checkObjectProps(expectedQuestionProps, payload[0]);
+        const secondObjectProps = checkObjectProps(expectedQuestionProps, payload[1]);
+        const thirdObjectProps = checkObjectProps(expectedQuestionProps, payload[2]);
+        const fourthObjectProps = checkObjectProps(expectedQuestionProps, payload[3]);
+
+        expect(payload.length).toBe(questionListFixtures.length);
+        expect(thirdObjectProps &&
+               secondObjectProps &&
+               firstObjectProps &&
+               fourthObjectProps).toBe(true);
+      });
+    });
+  });
+
+  describe('postScript: ', () => {
+    mock = new MockAdapter(axios);
+
+    beforeEach(() => {
+      mock.onPost('/scripts').reply(200, scriptFixture
+      );
+    });
+
+    afterEach(() => {
+      mock.reset();
+    });
+
+    describe('axios POST request: ', () => {
+      it('should add the destory action to the store and call goBack from history: ', () => {
+        const { body, description, name } = scriptFixture
+        const script = { body, description, name }
+        const expectedDestroyAction = destroy('ScriptForm');
+        const history = { goBack: jest.fn() };
+
+        return store.dispatch(postScript(script, questionListFixtures, history))
+          .then(() => {
+            const dispatchedActions = store.getActions();
+            const { type, meta } = dispatchedActions[0];
+
+            expect(dispatchedActions[0]).toEqual(expectedDestroyAction);
```

Comment:
The "internals" that are checked here are not explicitly set by us in the fixture on line 182 but are set by the code being tested, so it feels appropriate to check them.